InMobi/src/main/java/com/mopub/nativeads/InMobiNative.kt

```diff
@@ -0,0 +1,320 @@
+import android.os.Handler
+import android.os.Looper
+import android.util.TypedValue
+import android.view.View
+import android.view.ViewGroup
+import com.inmobi.ads.AdMetaInfo
+import com.inmobi.ads.InMobiAdRequestStatus
+import com.inmobi.ads.InMobiNative
+import com.inmobi.ads.exceptions.SdkNotInitializedException
+import com.inmobi.ads.listeners.NativeAdEventListener
+import com.inmobi.ads.listeners.VideoEventListener
+import com.mopub.common.DataKeys
+import com.mopub.common.logging.MoPubLog
+import com.mopub.common.logging.MoPubLog.AdapterLogEvent
+import com.mopub.mobileads.InMobiAdapterConfiguration
+import com.mopub.mobileads.InMobiBanner
+import java.util.*
+
+class InMobiNative : CustomEventNative() {
+
+    private var inMobiStaticNativeAd: InMobiNativeAd? = null
+    private var mPlacementId: Long = 0
+    private var mNativeListener: CustomEventNativeListener? = null
+
+    override fun loadNativeAd(
+        context: Context,
+        customEventNativeListener: CustomEventNativeListener,
+        localExtras: Map<String, Any>,
+        serverExtras: Map<String, String>
+    ) {
+        mNativeListener = customEventNativeListener
+
+        try {
+            mPlacementId = InMobiAdapterConfiguration.getPlacementId(serverExtras)
+        } catch (placementIdException: InMobiAdapterConfiguration.InMobiPlacementIdException) {
+            val errorMsg =
+                "InMobi Native request failed. Placement Id is not available or incorrect. Please make sure you set valid Placement Id on MoPub UI."
+            onError(customEventNativeListener, errorMsg)
+            return
+        }
+
+        InMobiAdapterConfiguration.initializeInMobi(
+            serverExtras,
+            context,
+            object : InMobiAdapterConfiguration.InMobiInitCompletionListener {
+                override fun onSuccess() {
+                    loadNative(context, serverExtras, customEventNativeListener)
+                }
+
+                override fun onFailure(error: Throwable) {
+                    val msg = "InMobi interstitial request failed due to InMobi initialization failed with an exception."
+                    onError(customEventNativeListener, msg)
+                }
+            })
+    }
+
+    private fun onError(customEventNativeListener: CustomEventNativeListener, errorMsg: String) {
+        customEventNativeListener.onNativeAdFailed(NativeErrorCode.NATIVE_ADAPTER_CONFIGURATION_ERROR)
+        MoPubLog.log(
+            getAdNetworkId(),
+            AdapterLogEvent.LOAD_FAILED,
+            TAG,
+            NativeErrorCode.NATIVE_ADAPTER_CONFIGURATION_ERROR,
+            errorMsg
+        )
+    }
+
+    private fun getAdNetworkId(): String {
+        return mPlacementId.toString() ?: ""
+    }
+
+    private fun loadNative(context: Context, serverExtras: Map<String, String>, customEventNativeListener: CustomEventNativeListener) {
+        inMobiStaticNativeAd = try {
+            if (context == null) {
```

Comment:
As `context` is marked as non-nullable this check isn't required.

Reply:
removed